### PR TITLE
Visual tests

### LINKED ISSUE #13: github actions for visual comparison
[Refer this experiment](https://github.com/hlxsites/vg-volvotrucks-us/pull/412/files) done for Volvo Trucks franklin project.

### PR DESCRIPTION
**Pull request by sdmcraft**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #13 

Test URLs:
- Before: https://main--sunstar--hlxsites.hlx.live/
- After: https://visual-tests--sunstar--hlxsites.hlx.live/

- [ ]  If you are adding a new block or a variation to an existing block, has it been added in the [block-library](https://adobe.sharepoint.com/:x:/r/sites/HelixProjects/Shared%20Documents/sites/sunstar/sunstar/sidekick/library.xlsx?d=w7f8a415fe192444caa87806b565bb62d&csf=1&web=1&e=oJWCS0) ?

### CODE REVIEW

**Comment by jindaliiita** [comment]
> ### 🔸 1 visual difference detected
> * **/** ([main](https://main--sunstar--hlxsites.hlx.page/) vs [branch](https://visual-tests--sunstar--hlxsites.hlx.live/))Expected an image 1280px by 3210px, received 1280px by 3727px. 742373 pixels (ratio 0.16 of all image pixels) are different.
> 
> The diff images are [attached in the artifact](https://github.com/hlxsites/sunstar/actions/runs/6206021424)

@sdmcraft I am not able to see any diff image as mentioned in the above comment

**Comment by sdmcraft** [comment]
> @sdmcraft I am not able to see any diff image as mentioned in the above comment

You'd need to download the test artifact and unzip it. 
image

BTW, there was a problem with footer handling and it should be addressed with https://github.com/hlxsites/sunstar/pull/27/commits/4b09b15725a0edfb32601f9fd5a37553cf65ddef

**Review by JiangLong2019 — APPROVED** [review]
It's a great tool which can help us a lot, please share with the team when you find time.

File: .github/workflows/visual-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Visual Tests
+
+on:
+  pull_request:
+    types: [ opened, synchronize ]
+
+env:
+  TEST_PATHS: "/"
+  TEST_PATHS_INDEXES: "/sidekick/library.json "
+  DOMAIN_MAIN: "main--sunstar--hlxsites.hlx.page"
```
Comment (jindaliiita) [inline .github/workflows/visual-tests.yaml:10]: This means we always have to provide .page url in the PR also.

File: .github/visual-tests/playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    actionTimeout: 0,
+    /* Base URL to use in actions like `await page.goto('/')`. */
+    // baseURL: 'http://localhost:3000',
+
+    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
+    trace: 'off',
+  },
+
+  /* Configure projects for major browsers */
+  projects: [
```
Comment (jindaliiita) [inline .github/visual-tests/playwright.config.ts:46]: Why we have configured only single browser here ?

**Comment by jindaliiita** [comment]
@sdmcraft Sorry for being late but left couple of comments here. May be u can check those.

**Pull request merged**